src/app/Http/Middleware/BackpackErrorViews.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Backpack\CRUD\app\Http\Middleware;
+
+use Closure;
+use Illuminate\Support\Str;
+
+class BackpackErrorViews
+{
+    public function handle($request, Closure $next)
```

Comment:
Oh wait. But that'll only happen when the middleware is used. So... it'll only happen on admin panel routes, indeed.

Reply:
I still feel uneasy about changing such an important Laravel config at runtime... Is there any way we can limit the scope or impact of this?

A) Any chance we can do this only when a renderable error is thrown? Not all the time?
B) Any chance we can use the `backpack_view()` helper instead of re-doing what it does, but at a much higher level (overriding the app config)?

Reply:
This does not override, this adds more options to the paths array. 
Making Backpack the first ones to look at, this will only be used inside admin panel routes as you said. 

It's done because how Laravel register the error paths: https://github.com/laravel/framework/blob/31b3d29cea6974674da5296a30442f4d803f28e2/src/Illuminate/Foundation/Exceptions/RegisterErrorViewPaths.php#L7

That's called when an http exception is rendered: https://github.com/laravel/framework/blob/31b3d29cea6974674da5296a30442f4d803f28e2/src/Illuminate/Foundation/Exceptions/Handler.php#LL640C35-L640C35

To answer A), we can revert this PR a bind to the `ExceptionHandlerContract` like I did before, and in that case we only register the paths when an error is actually thrown. Maybe instead of `bind` that has the drawbacks, we could use the resolving callbacks ? https://laravel.com/docs/10.x/container#container-events

```php
$this->app->resolving(ExceptionHandlerContract::class, function (ExceptionHandlerContract $class, Application $app) {
           // ...
        });
```
Would you consider this less intrusive than bind ? And better scoped than the current solution ?

Reply:
Yes that WOULD be less intrusive. Let's go back to binding. Sorry for taking you down this worse path Pedro 🙏

I also think that `resolving()` is less bad than `binding()` so let's use that.

Also. Instead of tagging onto the `ExceptionHandlerContract` can't we tag onto `RegisterErrorViewPaths`? If we bind/resolve that one, there's minimal impact, we can only affect other packages or classes that try to do exactly what we try to do, customize the error views paths.

Reply:
The `RegisterErrorViewPaths` is an invokable class, is manually instantiated, not resolved from the container. 

We can listen to resolve on `HttpExceptionInterface`, it's more scoped to http exceptions (includes aborts, find or fails etc). 

Not sure it will catch server errors, but I am fine leaving that for Laravel to handle, we just want nice errors for backpack pages anyway, if there is a server error, you are not going to reach any backpack page. 🤷 

Do we have a path forward here ? let's do this ? 💪

Reply:
Daaamn. Ok let's use `HttpExceptionInterface`.